pkg/processor/runtime/dotnetcore/Wrapper.cs

```diff
@@ -12,31 +12,90 @@
 //  See the License for the specific language governing permissions and
 //  limitations under the License.
 
-using System;
-using System.Collections.Generic;
-using System.Text;
 using Nuclio.Sdk;
+using System;
 
 namespace processor
 {
 
     public class Wrapper
     {
         private delegate object MethodDelegate(Context context, Event eve);
+        private delegate void InitContextDelegate(Context context);
         private static MethodDelegate methodDelegate;
+        private static string initContextFunctionName = "InitContext";
+        private Type methodType;
 
         private ISocketHandler socketHandler;
+        private Context context;
 
         public Wrapper(string dllPath, string typeName, string methodName, string socketPath)
         {
+            
             CreateTypeAndFunction(dllPath, typeName, methodName);
-            StartUnixSocketHandler(socketPath);
+
+            InitUnixSocketHandler(socketPath);
+
+            //Run the function's method that initializes the context
+            try
+            {
+                ExecuteInitContext();
+                socketHandler.SendMessage("{ 'kind': 'wrapperInitialized', 'attributes': {'ready': 'true'}");
+                context.Logger.Info("Wrapper ready.");
+
+            }
+            catch (Exception e)
+            {
+                context.Logger.Error(e.Message);
+                context.Logger.Error(e.StackTrace);
+                throw new Exception(e.Message);
+            }
+
+            StartUnixSocketHandler();
+
+        }
+
+        private void ExecuteInitContext()
+        {
+            if (methodType == null)
+            {
+                context.Logger.Error("Failed to execute InitContext(): type not found.");
+                throw new Exception("Failed to execute " + initContextFunctionName + ": type not found.");
+            }
+
+            context = new Context();
+            var initMethod = methodType.GetMethod(initContextFunctionName);
+            if (initMethod != null)
+            {
+                //invoke the method
+                var methodDelegate = (InitContextDelegate)Delegate.CreateDelegate(typeof(InitContextDelegate), 
+                    null, initMethod, false);
+                context.Logger.LogEvent += LogEvent;
+                if(methodDelegate != null)
+                {
+                    methodDelegate.Invoke(context);
+                }
+                else
+                {
+                    context.Logger.Error("Could not find method delegate for " + initContextFunctionName);
+                }
+                
+            }
+            else
+            {
+                context.Logger.Error("Could not locate method " + initContextFunctionName);
+            }
```

Comment:
most functions will probably wont have init context - i see it spammy and confusing